pandas/core/internals/blocks.py

```diff
@@ -273,6 +273,8 @@ def make_block_same_class(self, values, placement=None, ndim=None, dtype=None):
             )
         if placement is None:
             placement = self.mgr_locs
+        if ndim is None:
+            ndim = self.ndim
```

Comment:
Alternative option is to pass `self.ndim` to the `make_block_same_class` call in `Block.take_nd`, but this might be more general for other locations where `make_block_same_class` is called without specifying `ndim`. I don't directly can think of a case where you don't want to preserve the ndim unless it was specified.

And eg `make_block` just above always uses `self.ndim`

Reply:
This seems reasonable to me